### PR TITLE
Remove python 3.6 references

### PR DESCRIPTION
**Pull request by AnandInguva**
Fixes: https://github.com/apache/beam/issues/25002

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
R: @tvalentyn @aaltay

**Comment by AnandInguva** [comment]
Run Python Dataflow ValidatesRunner

File: .test-infra/tools/python_installer.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@
 set -euo pipefail
 
 # Variable containing the python versions to install
-python_versions_arr=("3.8.9" "3.6.13" "3.7.10" "3.9.4" "3.10.1")
+python_versions_arr=("3.7.16" "3.8.16" "3.9.16" "3.10.10")
```
Comment (tvalentyn) [inline .test-infra/tools/python_installer.sh:23]: whichever version is in the beginning will be the default Python intepreter (e.g. when one runs `python3`). This line will change default from 3.8 to 3.7. Let's keep it 3.8 or change to 3.9 / 3.10.

Comment (AnandInguva) [inline .test-infra/tools/python_installer.sh:23]:
```suggestion
python_versions_arr=("3.8.16" "3.7.16" "3.9.16" "3.10.10")
```

File: .test-infra/tools/README.md [hunk 2]
```diff
@@ -47,11 +47,11 @@ Example:
 - Original
 
 ```bash
-python_versions_arr=("3.6.13" "3.7.10" "3.8.9" "3.9.4")
+python_versions_arr=("3.7.16" "3.8.16" "3.9.16" "3.10.10")
```
Comment (AnandInguva) [inline .test-infra/tools/README.md:50]:
```suggestion
python_versions_arr=("3.8.16" "3.7.16" "3.9.16" "3.10.10")
```

File: .test-infra/tools/README.md [hunk 3]
```diff
@@ -47,11 +47,11 @@ Example:
...
 ```
 
 - Change
 
 ```bash
-python_versions_arr=("3.6.13" "3.7.10" "3.9.0" "3.10.2")
+python_versions_arr=("3.7.10" "3.9.0" "3.10.2")
```
Comment (AnandInguva) [inline .test-infra/tools/README.md:56]:
```suggestion
python_versions_arr=("3.8.10" "3.9.0" "3.10.2")
```

**Review by AnandInguva — COMMENTED** [review]
change py version order

File: .test-infra/tools/python_installer.sh (hunk 1 above)
Comment (AnandInguva) [reply .test-infra/tools/python_installer.sh]: Changed it. PTAL @tvalentyn

**Comment by tvalentyn** [comment]
LGTM

**Pull request merged**